Adicionando 3088 (#182)
petbccufscar/URI: Strings/3088/README.md

# Problema:

Em muitos idiomas uma regra deve ser seguida ao escrever qualquer texto digital : não use espaço em branco antes de uma vírgula ou um ponto .

Escreva um programa que, dado um texto de entrada, remove todo espaço que é imediatamente sucedido por uma vírgula ou por um ponto. Se houver mais do que um espaço que satisfaz esta condição, remova apenas um deles.

##### Link do problema: https://www.urionlinejudge.com.br/judge/pt/problems/view/3088
 
# Resolução:

> O exercício consiste em ler casos de teste ate acabar o arquivo, e para cada caso de teste teremos que retirar os espaços à esquerda do ',' e '.'.

Reviewer: ```suggestion
O exercício consiste em ler casos de teste até acabar o arquivo e, para cada caso de teste, teremos que retirar os espaços à esquerda do ',' e '.'.
```

Primeiro instanciamos as variáveis necessárias, sendo elas: um vetor do tipo `char` (onde armazenaremos a string lida) e um tipo `int` (que será nosso índice para imprimir cada `char`).

```c
    char string[1000];
```

Utilizaremos um `while()` para passar por todos os casos de teste. Dentro dele utilizaremos um `~scanf("%[^\n]%*c", string)`, que por questões de tempo essa leitura será mais rápida devido ao operador `~` (Bitwise complement) que trabalha diretamente com o valores binários, assim operando com mais desempenho.

```c
    char string[1000];
    while (~scanf("%[^\n]%*c", string))
    {
        \\código para cada caso individual
    }
```

> Dentro dele iremos zerar o indice (`i`, pois a cada leitura iremos recomecar as comparacoes) e em seguida utilizaremos outro `while(string[i])` para identificar o fim da frase, ou seja, quando nao existir nada no `string[i]` sairemos do `while`.

Reviewer: ```suggestion
Dentro dele iremos zerar o índice (`i`, pois a cada leitura iremos recomeçar as comparações) e em seguida utilizaremos outro `while(string[i])` para identificar o fim da frase, ou seja, quando não existir nada no `string[i]`, sairemos do `while`.
```

```c
    while (string[i])
        {
            \\código para comparar caractere por caractere
        }
```

> Agora iremos passar caractere por caractere para identificar se existe espaço (`' '`), se o encontrar-mos verificamos se o caractere seguinte é `,` *ou* `.`, se for imprimimos o caractere seguinte ao espaco e depois incrementamos o `i`, caso nao seja `,` *ou* `.` imprimimos o espaço e seguimos; caso nao seja espaço imprimimos o caractere em analise e seguimos; ao final do `while` interno sempre incrementamos o `i`. 

Reviewer: ```suggestion
Agora iremos passar caractere por caractere para identificar se existe espaço (`' '`), se o encontrarmos verificamos se o caractere seguinte é `,` *ou* `.`, se for imprimimos o caractere seguinte ao espaço e depois incrementamos o `i`, caso não seja `,` *ou* `.`, imprimimos o espaço e seguimos; caso não seja espaço, imprimimos o caractere em análise e seguimos; ao final do `while` interno sempre incrementamos o `i`.
```

```c
    if (string[i] == ' ')
        if (string[i + 1] == ',' || string[i + 1] == '.')
            putchar_unlocked(string[i + 1]), ++i;
        else
            putchar_unlocked(string[i]);
    else
        putchar_unlocked(string[i]);

    ++i;
```

> Para questao de desempenho tambem, utilizaremos o `putchar_unlocked`, pois ele é não bloqueante e nao utiliza buffs, logo, agilizando a impressao.

Reviewer: ```suggestion
Para questão de desempenho também, utilizaremos o `putchar_unlocked`, pois ele é não bloqueante e não utiliza buffs, agilizando a impressão.
```
